controllers/drplacementcontrol_controller.go

```diff
@@ -2422,3 +2447,183 @@ func (r *DRPlacementControlReconciler) determineDRPCState(
 
 	return AllowFailover, msg, nil
 }
+
+// ensureVRGsManagedByDRPC ensures that VRGs reported by ManagedClusterView are managed by the current instance of
+// DRPC. This is done using the DRPC UID annotation on the viewed VRG matching the current DRPC UID and if not
+// creating or updating the exisiting ManifestWork for the VRG.
+// Returns a bool indicating true if VRGs are managed by the current DRPC resource
+func ensureVRGsManagedByDRPC(
+	log logr.Logger,
+	mwu rmnutil.MWUtil,
+	vrgs map[string]*rmn.VolumeReplicationGroup,
+	drpc *rmn.DRPlacementControl,
+	vrgNamespace string,
+) bool {
+	ensured := true
+
+	for cluster, viewVRG := range vrgs {
+		if rmnutil.ResourceIsDeleted(viewVRG) {
+			log.Info("VRG reported by view undergoing deletion, during adoption",
+				"cluster", cluster, "namespace", viewVRG.Namespace, "name", viewVRG.Name)
+
+			continue
+		}
+
+		if viewVRG.GetAnnotations() != nil {
+			if v, ok := viewVRG.Annotations[DRPCUIDAnnotation]; ok && v == string(drpc.UID) {
+				continue
```

Comment:
This question is a corner case, so don't think about it too much.  What if the VRG reported that it contains the correct annotation, but the ManifestWork, somehow, does not exist? In other words, should we also make sure the MW for it exists?

Reply:
This is important, so with this PR if we create a VRG MW it is with the UID of DRPC always. This ensures that if we see a MCV reporting a VRG with the same UID this DRPC instance created the same.

The MW is not deleted unless for specific reasons, so missing an MW once it reports the DRPC UID is by choice (for example deleting VRG for VR post it reports Secondary or in the case of DRPC deletion).

Beyond the above, an external entity should have deleted the MW, which should not be usually the case.

At this point the intention is to ensure that we delivered the MW and not to ensure that the MW exists, because in the move to secondary for VR, we delete the MW and wait for the MCV to report it is deleted, in this interval the MW would be missing, and if we decide to recreate the MW we would loop on this deletion almost forever.